doc/python/performance.md

```diff
@@ -0,0 +1,311 @@
+    file_extension: .py
+    mimetype: text/x-python
+    name: python
+    nbconvert_exporter: python
+    pygments_lexer: ipython3
+    version: 3.11.10
+  plotly:
+    description: Recommendations for increased speed, improved interactivity, and
+      the ability to plot even more data!
+    display_as: basic
+    language: python
+    layout: base
+    name: High Performance Visualization
+    order: 14
+    permalink: python/performance/
+    redirect_from:
+    - python/webgl-vs-svg/
+    - python/datashader/
+    thumbnail: thumbnail/webgl.jpg
+---
+
+## DataFrame Types
+
+*New in Plotly.py version 6*
+
+Plotly Express natively supports various dataframe libraries, including pandas, Polars, and PyArrow. When building figures with Plotly Express, changing your dataframe library may help improve performance.
+
+In versions of Plotly.py prior to version 6, Plotly Express functions accepted non-pandas dataframes as input but used the [dataframe interchange protocol](https://data-apis.org/dataframe-protocol/latest/) or converted those dataframes to pandas internally.
+
+See [the Plotly Express Arguments page](/python/px-arguments) for full details on supported dataframe libraries.
+
+## NumPy and NumPy Convertible Arrays for Improved Performance
+
+*New in Plotly.py version 6*
+
+You can improve the performance of generating Plotly figures that use a large number of data points by passing data as NumPy arrays, or in a format that Plotly can convert easily to NumPy arrays, such as pandas and Polars Series or DataFrames. These formats will usually show better performance than passing data as a Python list.
+
+Plotly.py uses Plotly.js for rendering, which supports typed arrays. In Plotly.py, NumPy arrays and NumPy-convertible arrays are base64 encoded before being passed to Plotly.js for rendering.
+
+### Arrays and Data Types Supported
+
+The following types of objects in Python are supported for base64 encoding for rendering with Plotly.js.
+
+- NumPy `numpy.ndarray` objects.
+- pandas Index, pandas Series, Polars Series, and PyArrow Chunked Array objects.
+- When working with Plotly Express, pandas DataFrame, Polars DataFrame and PyArrow DataFrame objects passed to the `data_frame` argument of `px` functions.
+- Array objects that can be converted to `numpy.ndarray` objects, i.e., they implement `"__array__"` or `"__array_interface__"` and return a `numpy.ndarray`.
+
+The following [array data types](https://numpy.org/devdocs/reference/arrays.scalars.html) are supported:
+
+- float32
+- float64
+- int8
+- uint8
+- int16
+- uint16
+- int32
+- uint32
+
+*If the array dtype is **int64** or **uint64**, often the default dtype for arrays in NumPy when no dtype is specified, those dtypes will be changed to other types internally by Plotly.py where possible. When working with NumPY directly, you can [specify the `dtype`](https://numpy.org/doc/stable/user/basics.types.html#array-types-and-conversions-between-types) when creating `ndarray` objects.
```

Comment:
Slight improvement could be: 
When working with integers it is preferable to specify the `dtype` i.e. to avoid an extra step of finding the minimum and maximum of the list which by default uses Big-Integers (int64).